pyproject.toml

```diff
@@ -27,24 +26,42 @@ packages = [
     { include = "humanloop", from = "src"}
 ]
 
+
 [project.urls]
 Repository = 'https://github.com/humanloop/humanloop-python'
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = "^3.9"
 httpx = ">=0.21.2"
 httpx-sse = "0.4.0"
 pydantic = ">= 1.9.2"
 pydantic-core = "^2.18.2"
 typing_extensions = ">= 4.0.0"
+parse = "^1.20.2"
+opentelemetry-sdk = "^1.27.0"
+opentelemetry-api = "^1.27.0"
+opentelemetry-instrumentation-openai = "^0.33.3"
+opentelemetry-instrumentation-cohere = "^0.33.3"
+opentelemetry-instrumentation-anthropic = "^0.33.3"
+opentelemetry-instrumentation-groq = "^0.33.3"
+opentelemetry-instrumentation-replicate = "^0.33.3"
```

Comment:
these are probably too strict. 

the existing libraries are specified with `>=` which i think is a better choice than `^` (as if opentelemetry releases a v2 that is compatible around the relevant parts., the `^` would make it impossible to install with poetry. (overriding dependencies/making ignoring specific constraints when installing is extremely hard with poetry)

also these should be as loose as possible - we should look for the lowest reasonable version to support.

Reply:
(loosening these constraints may make python 3.8 possible to install again, so do check)

Reply:
I've checked the Instrumentor packages, and they impose a hard dependency on Python 3.9.

https://github.com/traceloop/openllmetry/blob/main/packages/opentelemetry-instrumentation-openai/pyproject.toml#L25

I've also checked their release history on PyPi and even the earliest version is tagged under Python 3.9

https://pypi.org/project/opentelemetry-instrumentation-openai/0.0.1/

<img width="321" alt="image" src="https://github.com/user-attachments/assets/2191df0a-e605-4c72-876b-4f8a8441d652">

Reply:
Please also action Harry's point around making them lowest reasonable version

Reply:
Still open: we need to discuss with gusto what Otel version is working. I have lowered the restrictions on dependencies

Reply:
i don't think the pypi tags are authoritative. i think they're literally just tags you can select for your package.

also, on the link you pasted, right above the bit you took a screenshot of, i see that it says `Python>=3.8.1`.

<img width="335" alt="image" src="https://github.com/user-attachments/assets/abc46119-c891-43ef-8091-20336950b9fc">

Reply:
Anything need to happen here?